setup.py

```diff
@@ -0,0 +1,32 @@
+import subprocess
+
+from setuptools import find_packages, setup
+
+
+def run_code_formatters():
+    for tool in ["isort .", "black .", "flake8 src/", "flake8 test/", "mdformat ."]:
+        print(f"running `{tool}`")
+        subprocess.run(tool, shell=True)
+
+
+if __name__ == "__main__":
+    setup(
+        name="cs101",
+        version="1.0",
+        url="https://github.com/jameshughes89/cs101",
+        python_requires=">=3.8",
+        packages=find_packages(),
+        install_requires=[
+            "black==22.1.0",
+            "flake8==4.0.1",
+            "flake8-black==0.3.2",
+            "flake8-isort==4.1.1",
+            "isort==5.10.1",
+            "mdformat==0.7.13",
+            "mdformat-gfm==0.3.5",
+            "mdformat-black==0.1.1",
+            "sphinx==4.0.1",
+            "sphinx-rtd-theme==1.0.0",
+        ],
+        entry_points={"console_scripts": [f"format = setup:{run_code_formatters.__name__}"]},
```

Comment:
I spent a crazy amount of time trying to, but setup tools was pissy about sphinx not existing. Similarly with flake8, which is strange because it worked in the formatter script.

Reply:
That's because sphinx isn't available at build time. The reason the formatters work is you're not actually referencing them, just firing up a shell that will eventually reference them. You could use the same workaround, or add sphinx as a build dependency in `pyproject.toml`, similar to how [duckbot installs nltk data](https://github.com/duck-dynasty/duckbot/blob/main/setup.py#L15-L25).